block/manager.go

```diff
@@ -118,6 +122,18 @@ func NewManager(
 	blockStore *goheaderstore.Store[*types.Block],
 ) (*Manager, error) {
 	s, err := getInitialState(store, genesis)
+	// genesis should have exactly one "validator", the centralized sequencer.
+	// this should have been validated in the above call to getInitialState.
+	sequencer := &cmtypes.Validator{
```

Comment:
can you make use of `GetValidatorSetFromGenesis` here to avoid code duplication?